minecraft/protocol/resource_pack.go

```diff
@@ -1,30 +1,57 @@
 	UUID string
-	// Version is the version of the resource pack. The client will cache resource packs sent by the server as
-	// long as they carry the same version. Sending a resource pack with a different version than previously
+	// Version is the version of the behaviour pack. The client will cache behaviour packs sent by the server as
+	// long as they carry the same version. Sending a behaviour pack with a different version than previously
 	// will force the client to re-download it.
 	Version string
-	// Size is the total size in bytes that the resource pack occupies. This is the size of the compressed
-	// archive (zip) of the resource pack.
+	// Size is the total size in bytes that the behaviour pack occupies. This is the size of the compressed
+	// archive (zip) of the behaviour pack.
 	Size uint64
-	// ContentKey is the key used to decrypt the resource pack if it is encrypted. This is generally the case
-	// for marketplace resource packs.
+	// ContentKey is the key used to decrypt the behaviour pack if it is encrypted. This is generally the case
+	// for marketplace behaviour packs.
 	ContentKey string
 	// SubPackName ...
 	SubPackName string
 	// ContentIdentity ...
 	ContentIdentity string
-	// HasScripts specifies if the resource packs has any scripts in it. A client will only download the
-	// resource pack if it supports scripts, which, up to 1.11, only includes Windows 10.
+	// HasScripts specifies if the behaviour packs has any scripts in it. A client will only download the
+	// behaviour pack if it supports scripts, which, up to 1.11, only includes Windows 10.
 	HasScripts bool
 }
 
+// TexturePackInfo represents a texture pack's info sent over network. It holds information about the
+// texture pack such as its name, description and version.
+type TexturePackInfo struct {
+	// UUID is the UUID of the texture pack. Each texture pack downloaded must have a different UUID in
+	// order for the client to be able to handle them properly.
+	UUID string
+	// Version is the version of the texture pack. The client will cache texture packs sent by the server as
+	// long as they carry the same version. Sending a texture pack with a different version than previously
+	// will force the client to re-download it.
+	Version string
+	// Size is the total size in bytes that the texture pack occupies. This is the size of the compressed
+	// archive (zip) of the texture pack.
+	Size uint64
+	// ContentKey is the key used to decrypt the behaviour pack if it is encrypted. This is generally the case
+	// for marketplace texture packs.
+	ContentKey string
+	// SubPackName ...
+	SubPackName string
+	// ContentIdentity ...
+	ContentIdentity string
+	// HasScripts specifies if the texture packs has any scripts in it. A client will only download the
+	// behaviour pack if it supports scripts, which, up to 1.11, only includes Windows 10.
+	HasScripts bool
+	// RTXEnabled ...
+	RTXEnabled bool
```

Comment:
This bool can use documentation. Resource packs can be RTX resource packs so that ray tracing may be enabled when they're applied. This specifies if the pack is one of those.